### DISCUSSION TITLE
Enabling the latest version causes CTRL+RIGHT arrow to stop working

### DISCUSSION THREAD

**Discussion by jjcampos** [Genuitec/Copilot4Eclipse-community#42] [Help] [answered]
Hi there, 

Today I've update copilot from the version I was in to the latest version (Unfortunately I don't know what version I was previously but it was at least from 4 months back, when I first started using it in Eclipse Enterprise IDE).

The version I'm using is: 
Genuitec, LLC	Copilot4Eclipse	1.3.1.202410261755	com.genuitec.copilot4eclipse.feature

As soon as I installed this version opening any java file causes the CTRL+right arrow to stop working. This shortcut is used to jump the cursor from word to word or depending on your formatting camel case words as well.

The way I've found that the issue is the copilot plugin was that I first upgraded from eclipse 2024-03 to 2024-06 and together with it I updated all the other plugins and libraries with it, copilot one was listed there.

Once I identified the bug, I then tried to upgrade it again to 2024-09, the bug still persisted.
Then I decided to test running a clean version of Eclipse version 2024-06, it worked fine, I then started making the changes to run the IDE with my preferences and on each change I was not seeing the bug.

The moment I installed Copilot the bug shows up, cannot use CTRL+right arrow anymore. 
Just using the "Disable completions" options from the Copilot plugin icon on the IDE, makes the CTRL+right arrow work again.

This same version of copilot plugin when completions are Enabled causes any open SQL file to generate an error.
```
An error has occurred. See error log for more details.
Cannot invoke "java.net.URI.toString()" because "fileUri" is null
```
I've attached a screenshot for it.

The company I work for has a enterprise license for Copilot, let me know if I need to open this support in a different medium or with a different account please.
![Selection_083](https://github.com/user-attachments/assets/cd0c4088-e5ad-4cb3-b230-9189acc72ded)

I just found out from another thread that it also breaks CTRL + H (and testing it with this version), the Eclipse version I downloaded for linux were:
eclipse-jee-2024-06-R-linux-gtk-x86_64.tar.gz
and
eclipse-jee-2024-03-R-linux-gtk-x86_64.tar.gz

**Comment by wayneparrott** [comment, marked as answer]
Thanks for the detailed report.

ctl+right is bound to 2 commands as shown below. Copilot4Eclipse uses this keybinding when a ghost-text completion is displayed to incrementally accept the completion 1 word at a time. The "Accept Next Word" command is only active in the "completion" context. It seems that context's activation is goofing up across some Eclipse versions. The Copilot4Eclipse ctl+right binding should only be active when a copilot completion is being displayed, i.e., the "completion" context. I have opened a ticket for the team to investigate this.

![image](https://github.com/user-attachments/assets/7cb724e4-0f36-4b4e-a53f-4ca030222043)

In the meantime you can disable or reassign the Copilot4Eclipse "Accept Next Word" command. 
1. Open the Keys preferences:  select Copilot4Eclipse menu > Preferences > General > Keys
2. Sort bindings by Category and scroll to the Copilot4Eclipse category of key bindings.
3. Select the "Accept Next Word" binding and either the "Unbind Command" to remove it from your keys bindings or change the Binding field to a different set of keys.
4. Select "Apply" button

Doing this should allow the "Next Word" command to resume operation as well as Copilot4Eclipse suggestions without the Accept Next Word key binding.

I'll look into the SQL Editor asap. I think I know what's goofed but need to research it fully.

Lastly you referenced a Search issue. Certain Eclipse distros were affected by Copilot4Eclipse 1.2.1. That was fixed by the 1.2.2 version.
Does Search NOT work for you with Copilot4Eclipse 1.3.1 installed?

Wayne
Copilot4Eclipse Team

**Reply by jjcampos** [reply]
Hi @wayneparrott 

Thank you for your fast reply! 

Your suggestion on checking the Key Bindings was dead correct. That function "Accept Next Word" had the binding assigned to the CTRL+Right. I just un-binded it and it is now working fine, thank you for that.

The SQL file error is still happening on Eclipse 2024-03 and 2024-06, thanks for looking into it.

The search issue I mentioned was not really the search function, it was the key binding to it, CTRL+H strangely after removing the binding to "Accept Next Word" it is now working fine on Eclipse 2024-03 and 2024-06. I even tried to bind the key again and problem did not happen. So it might have been a strange combination of events that fixed it. The search function still worked fine, just the binding seems broken, I apologize if it was not clear.

A note, "CTRL+right" is a universal text editor shortcut on *nix / windows systems (excluded macos for this specific one), so, if I may, I would suggest to not use these universal editing shortcuts as key bindings, here is a link I found with a list of them:
https://www.howtogeek.com/115664/42-text-editing-keyboard-shortcuts-that-work-almost-everywhere/

Best,
Jorge

**Reply by wayneparrott** [reply]
re: SQLEditor error, I have replicated the issue you shared. We are researching to understand what's goofed. 

W

**Reply by wayneparrott** [reply]
@jjcampos The latest 1.3.2 update retains the Copilot4Eclipse keybindings. But I discovered that the "completion" context is incorrectly prematurely being activated. Thus, the negative experience with CTL+Right not nav'ing to the next word. How it should work is the completion context is supposed to only be active when a copilot completion (ghost-text) is displayed which is immediately to the right of the cursor. Only in that limited context should CTL+Right ever be rebound. Once the completion is closed or fully accepted the context should immediately return to the normal text edit context. We will fix this soon. For those that dislike this default we will provide some alternatives in a future update. Appreciate your feedback.
W

**Reply by jjcampos** [reply]
Thank you for the explanation.

Looking forward to the next updates. Hoping the SQL file problem is fixed on the next updates as well.

Best,
Jorge

**Reply by jjcampos** [reply]
Hey @wayneparrott I hope everything is fine there.

So today I updated CoPilot on my Eclipse IDE (I'm using Version: 2024-09 (4.33.0); Build id: 20240905-0614)
The version I'm using now is: 
Genuitec, LLC	Copilot4Eclipse	1.3.2.202411110916	com.genuitec.copilot4eclipse.feature

With this version the SQL file problem went away but it no longer supports SQL file? (and there is a typo on the message saying so)

![copilot_sql_file_problem](https://github.com/user-attachments/assets/894bd19c-6592-44d4-a210-ccd0b312c49b)


Best,
Jorge

**Reply by wayneparrott** [reply]
Unfortunately the DTP SQL Editor is one of a few Eclipse editors with an incompatible implementation for Copilot4Eclipse. I'm hoping to look into the details more closely over the holidays. Perhaps we can find a workaround.  I'll post an update once I have had more time for a deep investigation.

Wayne
Copilot4Eclipse Team

**Reply by jjcampos** [reply]
Sounds good! Thanks for the feedback!

Best,
Jorge
